src/ComboBoxStateful/index.jsx

```diff
@@ -0,0 +1,668 @@
+
+        this.setState( prevState =>
+        {
+            const activeOption = getOption(
+                id,
+                prevState.flatOptions,
+            ).id;
+
+            return { activeOption };
+        } );
+    }
+
+    handleOnFocus( e )
+    {
+        const callback = this.props.onFocus;
+
+        if ( callback )
+        {
+            callback( e );
+        }
+
+        this.setState( {
+            isFocused : true,
+            isOpen    : true,
+        } );
+    }
+
+    handleOnBlur( e )
+    {
+        const callback = this.props.onBlur;
+
+        if ( callback )
+        {
+            callback( e );
+        }
+
+        this.setState( {
+            activeOption    : undefined,
+            isFocused       : false,
+            isOpen          : false,
+            filteredOptions : undefined,
+            searchValue     : undefined,
+        } );
+    }
+
+    render()
+    {
+        const {
+            buttonIsReadOnly,
+            className,
+            dropdownPlaceholder,
+            dropdownPosition,
+            forceHover,
+            hasError,
+            iconButtonIsDisabled,
+            iconPosition,
+            iconType,
+            inputPlaceholder,
+            isDisabled,
+            isMultiselect,
+            isReadOnly,
+            isSearchable,
+            onKeyPress,
+            onKeyUp,
+            onMouseOut,
+            onMouseOutIcon,
+            onMouseOver,
+            onMouseOverIcon,
+            onScroll,
+            options,
+            textAlign,
+        } = this.props;
+
+        const {
+            activeOption,
+            filteredOptions,
+            flatOptions,
+            id,
+            isFocused,
+            isOpen,
+            searchValue,
+            selection,
+        } = this.state;
+
+        const optionVal = getOption( selection, flatOptions ) ?
+            getOption( selection, flatOptions ).text : undefined;
+
+        const optionsToShow = filteredOptions ?
+            this.optionsFilter( filteredOptions ) : options;
+
+        return (
+            <ComboBox
+                activeOption         = { activeOption }
+                buttonIsReadOnly     = { buttonIsReadOnly }
+                className            = { className }
+                dropdownPlaceholder  = { dropdownPlaceholder }
+                dropdownPosition     = { dropdownPosition }
+                forceHover           = { forceHover }
+                hasError             = { hasError }
+                iconButtonIsDisabled = { iconButtonIsDisabled }
+                iconPosition         = { iconPosition }
+                iconType             = { iconType ||
+                    ( isOpen ? 'up' : 'down' ) }
+                id                   = { id }
+                inputIsReadOnly      = { !isSearchable || !isOpen }
+                inputRef             = { ( ref ) =>
+                {
+                    this.inputRef = ref;
+                } }
```

Comment:
The above can fit on one line:
```
inputRef = { ref => this.inputRef = ref }
```

Reply:
Hmm you may have fooled the linter but that arrow function is still returning assignment 😜😃

I was gonna say that that linter rule is nonsense but actually there is a certain point to it: 

Since this is a class component, here should be a class method (e.g `setInputRef`) that does the assignment:

```
inputRef = { this.setInputRef }
```

Reply:
Yeah you’re right. After drinking a morning coffee I see that the first part of my comment above is nonsense 🙈

I stand by the bit about the class method tho 😃